<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="mxfda">
<title>Functional Regression with Spatial Transcriptomics Data • mxfda</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<script src="../../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../../pkgdown.js"></script><meta property="og:title" content="Functional Regression with Spatial Transcriptomics Data">
<meta property="og:description" content="mxfda">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../../index.html">mxfda</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.2-1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../../articles/mx_fda.html">Defining an mxFDA object</a>
    <a class="dropdown-item" href="../../articles/mx_fpca.html">Functional principal component analysis for spatial summary functions</a>
    <a class="dropdown-item" href="../../articles/mx_funreg.html">Functional regression with spatial summary functions as covariates</a>
    <a class="dropdown-item" href="../../articles/web_only/st_fpca.html">Functional Regression with Spatial Transcriptomics Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/julia-wrobel/mxfda/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../../logo.png" class="logo" alt=""><h1>Functional Regression with Spatial Transcriptomics Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/julia-wrobel/mxfda/blob/HEAD/vignettes/web_only/st_fpca.Rmd" class="external-link"><code>vignettes/web_only/st_fpca.Rmd</code></a></small>
      <div class="d-none name"><code>st_fpca.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.github.io/seurat-object/" class="external-link">SeuratObject</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/julia-wrobel/mxfda/" class="external-link">mxfda</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/" class="external-link">ggpubr</a></span><span class="op">)</span></span></code></pre></div>
<p>This vignette will show how to use the <a href="https://github.com/julia-wrobel/mxfda/" class="external-link">mxfda</a> to
analyze the clustering of cells from spatial transcriptomics. The data
comes from a single cell spatial transcriptomics project exploring
differences in the tumor microenvironment that is associated with
immunotherapy (<span class="citation">Soupir et al. (2023)</span>). A
preprint of the manuscript ‘Increased spatial coupling of integrin and
collagen IV in the immunoresistant clear cell renal cell carcinoma tumor
microenvironment’ can be found here: <a href="https://www.biorxiv.org/content/10.1101/2023.11.16.567457v1" class="external-link uri">https://www.biorxiv.org/content/10.1101/2023.11.16.567457v1</a>.
Data to follow along with this vignette can be downloaded here: <a href="https://zenodo.org/doi/10.5281/zenodo.12730226" class="external-link uri">https://zenodo.org/doi/10.5281/zenodo.12730226</a>.</p>
<div class="section level2">
<h2 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h2>
<p>Kidney cancer (renal cell carcinoma; RCC) is one of the most common
cancer types, with around 5% of people getting it at some point in their
lives. Clear cell RCC (ccRCC) is the most common subtype of RCC and
typically has a fair initial response to treatments called immune
checkpoint inhibitors (ICI) or immunotherapy (IO). However, it’s common
for the tumors to develop resistance to these treatments over time,
usually 1 to 2 years. This data was generated to dive into understanding
what changes are happening in the microenvironment of the tumors that
may either lead to, or be a result of, developing treatment
resistance.</p>
<p>The data is generated with the Nanostring CosMx SMI platform,
providing us with information about the expression of ~960 genes, cell
locations, and a few immunofluorescence markers. Tissues were used to
create tissue microarrays (TMAs) - a grid of ~1mm cores allowing for
many samples to be placed onto a single slide - in pairs of tumor and
stroma (in the <code>annotation</code> column of the Seurat object).
Cell types were created using Nanostring’s single-cell annotation
package called <a href="https://github.com/Nanostring-Biostats/InSituType" class="external-link">InSityType</a>,
which uses <a href="https://github.com/Nanostring-Biostats/CellProfileLibrary" class="external-link">CellProfileLibrary</a>
references to identify cell types using a Bayesian approach (<span class="citation">Danaher et al. (2022)</span>). For the cell types that
are provided in the Seurat object, a semi-supervised approach meaning
cells in the samples were fit best to a cell type in the
CellProfileLIbrary reference and if there wasn’t a great fit (because of
something like cancer cell or the cell type isn’t in the reference)
it’ll be classified by a letter like ‘a’, ‘b’, etc. This is nice if it’s
known that a sample will have cell types that aren’t in the
reference.</p>
<p>In this vignette, we don’t have clinical information but we have
tissue annotations letting us know which fields of view (FOVs), or
samples, are from “Tumor” regions and “Stroma” regions. Univariate
clustering will be calculated for CD8 T cells with <a href="https://github.com/julia-wrobel/mxfda/" class="external-link">mxfda</a>.
Associations can performed with the annotated compartments that the
samples were from to assess if there is CD8 T cell clustering
differences between the “Tumor” or “Stroma” compartmetns.</p>
</div>
<div class="section level2">
<h2 id="preparing-the-data">Preparing the Data<a class="anchor" aria-label="anchor" href="#preparing-the-data"></a>
</h2>
<p>Data in the form of a Seurat object (v4) was downloaded from Zenodo
above and placed into a project folder. From the Seurat object, will
extract the <code>meta.data</code> as well as cell locations. Some QC
will need to be done to remove FOVs with low number of cells, otherwise
expression QC was previously performed.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read from disk - much faster than reading from URL but can be done</span></span>
<span><span class="va">kidney_st</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../../../seurat_object.Rds"</span><span class="op">)</span></span>
<span><span class="co">#the cell level features from the meta.data</span></span>
<span><span class="va">dat</span> <span class="op">=</span> <span class="va">kidney_st</span><span class="op">@</span><span class="va">meta.data</span></span></code></pre></div>
<p>To extract the cell locations, it will be slightly more complicated
as each slide (RCC3, RCC4, and RCC5) have their cell locations stored in
different <code>FOV</code> objects in the <code>kidney_st@images</code>
list. Will implement a <code>lapply</code> to iterate over each and then
return the centroids.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#create dataframe with the global cell centroids</span></span>
<span><span class="co">#iterate over each of the FOV objects</span></span>
<span><span class="va">centroids</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">kidney_st</span><span class="op">@</span><span class="va">images</span>, <span class="kw">function</span><span class="op">(</span><span class="va">slide</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="co">#Get the coordinates</span></span>
<span>  <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/GetTissueCoordinates.html" class="external-link">GetTissueCoordinates</a></span><span class="op">(</span><span class="va">slide</span><span class="op">@</span><span class="va">boundaries</span><span class="op">$</span><span class="va">centroids</span><span class="op">)</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="co">#keep only those that passed filtering from creating the seurat object</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">cell</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">dat</span><span class="op">$</span><span class="va">id</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co">#since there are 3 lists, bind them all together</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">bind_rows</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co">#column in the centroids with the cell identifier is called 'cell', rename to 'id'</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"id"</span> <span class="op">=</span> <span class="va">cell</span><span class="op">)</span></span>
<span><span class="co">#merge the centroids back to the meta.data</span></span>
<span><span class="va">dat</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">full_join</a></span><span class="op">(</span><span class="va">dat</span>,</span>
<span>                     <span class="va">centroids</span>,</span>
<span>                     by <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/join_by.html" class="external-link">join_by</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Since there may be some FOVs that have low cell counts, we can remove
those. Considering there are about 30 different cell types and we want
to calculate univariate clustering a good number of cells at a minimum
should be ~500. This removes a single bad quality FOV on Slide RCC5 -
FOV 13 - that only has 5 cells present.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">=</span> <span class="va">dat</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co">#group by slide and fov to calculate number of cells per slide</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">slide_ID_numeric</span>, <span class="va">fov</span><span class="op">)</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co">#remove FOVs with low number of cells</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">500</span><span class="op">)</span></span></code></pre></div>
<p>What remains are FOVs that have a fair number of cells (&gt;500) and
are from the tumor compartment. FOV 1 on RCC3 can be plotted to see how
large the sample is (in pixels) to get a fair idea of how far our
spatial summary functions should go.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">fov</span> <span class="op">==</span> <span class="fl">1</span>, <span class="va">tissue</span> <span class="op">==</span> <span class="st">"RCC3"</span><span class="op">)</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">x</span>, y <span class="op">=</span> <span class="va">y</span>, color <span class="op">=</span> <span class="va">nb_clus</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_equal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"RCC3 - FOV1"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>plot.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>color <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="st_fpca_files/figure-html/unnamed-chunk-5-1.png" width="1440"></p>
</div>
<div class="section level2">
<h2 id="spatial-analysis-with-mxfda">Spatial Analysis with <code>mxfda</code><a class="anchor" aria-label="anchor" href="#spatial-analysis-with-mxfda"></a>
</h2>
<p>For an <code>mxFDA</code> object the data needs to be broken up in to
a <code>spatial</code> component and a <code>metadata</code> component.
Important to note that it is also needed that the metadata needs 2
columns that would be a sample ID (such as a unique identifier for each
slide-fov combination) and subject ID (think of patient level if there
were multiple slide-fov combinations per patient). the metadata should
also contain any covariates or information that you would to later model
- since we want to see differences between the tumor and stromal
compartments we will also extract that column. For this data will use
the slide-fov combination for both the sample and subject assuming each
sample is independent.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#add unique slide-fov column to identify samples</span></span>
<span><span class="va">dat</span> <span class="op">=</span> <span class="va">dat</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sampleID <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">Slide_name</span>, <span class="st">"_"</span>, <span class="va">fov</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#select the new sample ID column and duplicate it for subject ID</span></span>
<span><span class="va">metadata</span> <span class="op">=</span> <span class="va">dat</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sampleID</span>, <span class="va">annotation</span><span class="op">)</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>subjectID <span class="op">=</span> <span class="va">sampleID</span>,</span>
<span>         <span class="co">#convert the annotation column to factor to make modeling easier later</span></span>
<span>         tissue <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#select the columns that will will tell what sample a cell is from, as well as location and type</span></span>
<span><span class="va">spatial</span> <span class="op">=</span> <span class="va">dat</span> <span class="op"><a href="../../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sampleID</span>, <span class="va">x</span>, <span class="va">y</span>, <span class="va">nb_clus</span>, <span class="va">annotation</span><span class="op">)</span></span>
<span><span class="co">#create the object for use in mxfda functions</span></span>
<span><span class="va">mxFDAobject</span> <span class="op">=</span> <span class="fu"><a href="../../reference/make_mxfda.html">make_mxfda</a></span><span class="op">(</span>metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>                         spatial <span class="op">=</span> <span class="va">spatial</span>,</span>
<span>                         subject_key <span class="op">=</span> <span class="st">"subjectID"</span>,</span>
<span>                         sample_key <span class="op">=</span> <span class="st">"sampleID"</span><span class="op">)</span></span>
<span><span class="va">mxFDAobject</span></span>
<span><span class="co">#&gt; mxFDA Object:</span></span>
<span><span class="co">#&gt;  Subjects: 64</span></span>
<span><span class="co">#&gt;  Samples: 64</span></span>
<span><span class="co">#&gt;  Has spatial data</span></span>
<span><span class="co">#&gt;  Univariate Summaries: None</span></span>
<span><span class="co">#&gt;  Bivariate Summaries: None</span></span>
<span><span class="co">#&gt;  Multivariate Summaries: None</span></span>
<span><span class="co">#&gt; FPCs not yet calculated</span></span>
<span><span class="co">#&gt; MFPCs not yet calculated</span></span>
<span><span class="co">#&gt; FCMs not yet calculated</span></span>
<span><span class="co">#&gt; MFCMs not yet calculated</span></span>
<span><span class="co">#&gt; Scalar on Functional Regression not calculated</span></span></code></pre></div>
<p>The spatial summary function to be extracted from the data is nearest
neighbor G, which measures the proportion of cells of interest (here CD8
T cells) that have a nearest neighbor some radius <em>r</em> or less.
The spatial summary function can be extracted using the function
<code><a href="../../reference/extract_summary_functions.html">extract_summary_functions()</a></code> specifying that we are
interested in the univariate methods
(<code>extract_func = univariate</code>), specifically the nearest
neighbor G function (<code>summary_func = Gest</code>). The search
radius around cells will range from 0 to 500 using the
<code>r_vec</code> option and edge correction will be done using the
reduced sampling method (<code>edge_correction = rs</code>).</p>
<p>If there are large regions of necrosis, connective tissue, of other
technical artifacts that cause large areas of samples to be missing
cells, the option <code>permute_CSR</code> should be used (<span class="citation">Wilson et al. (2022)</span>). This is usually more of
an issue with TMAs but does greatly depend on the tissue being
profiled.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mxFDAobject</span> <span class="op">=</span> <span class="fu"><a href="../../reference/extract_summary_functions.html">extract_summary_functions</a></span><span class="op">(</span><span class="va">mxFDAobject</span>,</span>
<span>                                        extract_func <span class="op">=</span> <span class="va">univariate</span>,</span>
<span>                                        summary_func <span class="op">=</span> <span class="va">Gest</span>,</span>
<span>                                        r_vec <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">500</span>,</span>
<span>                                        edge_correction <span class="op">=</span> <span class="st">"rs"</span>,</span>
<span>                                        <span class="co">#column with our cell annotations</span></span>
<span>                                        markvar <span class="op">=</span> <span class="st">"nb_clus"</span>,</span>
<span>                                        <span class="co">#cell type of interest</span></span>
<span>                                        mark1 <span class="op">=</span> <span class="st">"CD8.T.cell"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using Theoretical Complete Spatial Randomness for Nearest Neighbor G</span></span></code></pre></div>
<p>Calculated nearest neighbor G for our 0 to 500 can be visualized
using the S4 <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> function on the <code>mxFDA</code>
object. This function wraps <a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a> so other plot
modifiers can be used/applied to the output object as would be done with
any other <code>ggplot</code> object. The returned values from
<code>extract_summary_functions</code> contains a column of the observed
spatial summary function value <code>r</code>, a column of the CSR
estimate (whether permuted or theoretical) <code>csr</code>, and one for
the difference between observed and CSR <code>fundiff</code>. If the
spatial summary function was produced by a different software or
package, <code><a href="../../reference/add_summary_function.html">add_summary_function()</a></code> can be used to import to
the <code>mxFDA</code> object and the above column names will be
different. Here, the column of interest when plotting and further
analysis will be <code>fundiff</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mxFDAobject</span>, y <span class="op">=</span> <span class="st">"fundiff"</span>, what <span class="op">=</span> <span class="st">"uni g"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="st_fpca_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Functional principal component analysis (FPCA) allows researchers to
extract out information from a set of curves such as the population mean
(overall average curve shape), deviations from that mean that represent
the largest variation, and how much those deviations contribute to each
samples measured curve. This can be done with the
<code><a href="../../reference/run_fpca.html">run_fpca()</a></code> function in <a href="https://github.com/julia-wrobel/mxfda/" class="external-link">mxfda</a> . The
<code>mxFDA</code> object is passed in along with the spatial summary
function of interest (<code>"uni g"</code>) and 2 column names: one for
the column with radius values (<code>"r"</code>) and the other with
values relating to the spatial summary function
(<code>"fundiff"</code>). Lastly, <code>pve</code> can be used to select
the percent variance explained by the FPCs kept in the output.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mxFDAobject</span> <span class="op">=</span> <span class="fu"><a href="../../reference/run_fpca.html">run_fpca</a></span><span class="op">(</span><span class="va">mxFDAobject</span>,</span>
<span>                       metric <span class="op">=</span> <span class="st">"uni g"</span>,</span>
<span>                       r <span class="op">=</span> <span class="st">"r"</span>,</span>
<span>                       value <span class="op">=</span> <span class="st">"fundiff"</span>,</span>
<span>                       pve <span class="op">=</span> <span class="fl">0.99</span><span class="op">)</span></span>
<span><span class="co">#&gt; 64 sample have &gt;= 4 values for FPCA; removing 0 samples</span></span>
<span><span class="co">#&gt; Warning in sqrt(Eigen$values): NaNs produced</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">mxFDAobject</span><span class="op">)</span></span>
<span><span class="co">#&gt; mxFDA Object:</span></span>
<span><span class="co">#&gt;  Subjects: 64</span></span>
<span><span class="co">#&gt;  Samples: 64</span></span>
<span><span class="co">#&gt;  Has spatial data</span></span>
<span><span class="co">#&gt;  Univariate Summaries: Gest</span></span>
<span><span class="co">#&gt;  Bivariate Summaries: None</span></span>
<span><span class="co">#&gt;  Multivariate Summaries: None</span></span>
<span><span class="co">#&gt; FPCs Calculated:</span></span>
<span><span class="co">#&gt;  Gest: 12 FPCs describe 99% variance</span></span>
<span><span class="co">#&gt; MFPCs not yet calculated</span></span>
<span><span class="co">#&gt; FCMs not yet calculated</span></span>
<span><span class="co">#&gt; MFCMs not yet calculated</span></span>
<span><span class="co">#&gt; Scalar on Functional Regression not calculated</span></span></code></pre></div>
<p>Calling the summary of the <code>mxFDA</code> object returns 12 FPCs
were needed to explain 99% of the variance in our data. Again, using the
plotting function the first 2 FPCs can be plotted to show the mean and
direction of the first 2 functional principal components which explain
the largest variance between samples.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mxFDAobject</span>, what <span class="op">=</span> <span class="st">'uni g fpca'</span>, pc_choice <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mxFDAobject</span>, what <span class="op">=</span> <span class="st">'uni g fpca'</span>, pc_choice <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggarrange.html" class="external-link">ggarrange</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, nrow <span class="op">=</span> <span class="fl">1</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="st_fpca_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>Here we see the first and second functional principal components. The
black line is the mean of all of the samples, the blue line is +1
standard deviation in the direction of the variance explained by that
component and red line is -1 standard deviation. For FPC1 we interpret
the pattern of this FPC as a shift up or down from the mean of all
samples. Red are those with negative FPC values extracted from their
G(r) curves and blue are those with positive FPC values. For example,
high FPC corresponds to a negative deviation from the mean that moves
further from the mean from ~r = 100 to r = 400. We see that these shapes
deviating from the sample average describe ~58% of the variation in our
samples.</p>
<p>Next we can run a linear functional cox model (lfcm) for the
G(<em>r</em>) functions to check associations of CD8 T cell clustering
curves and whether those CD8 T cells are on a tumor FOV or a stroma FOV.
Since we don’t have an event like we do with survival (patient death) we
will use “Scalar on Function Regression” which will model our functional
data against some binary outcome like tissue source (tumor/stroma). This
function from <a href="https://github.com/julia-wrobel/mxfda/" class="external-link">mxfda</a> is called <code><a href="../../reference/run_sofr.html">run_sofr()</a></code> and
takes the <code>mxFDA</code> object, a name to store the model with in
the object, a formula to model (since we don’t have any useful
covariates will model functional data against the outcome
<code>tissue</code> class), the model type we want to fit, metric,
radius column name, and column name to model in the spatial summary
data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mxFDAobject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../../reference/run_sofr.html">run_sofr</a></span><span class="op">(</span><span class="va">mxFDAobject</span>, </span>
<span>                        model_name <span class="op">=</span> <span class="st">"fit_sofr_tissue"</span>, </span>
<span>                        formula <span class="op">=</span> <span class="va">tissue</span> <span class="op">~</span> <span class="fl">1</span>, </span>
<span>                        family <span class="op">=</span> <span class="st">"binomial"</span>,</span>
<span>                        metric <span class="op">=</span> <span class="st">"uni g"</span>, r <span class="op">=</span> <span class="st">"r"</span>, value <span class="op">=</span> <span class="st">"fundiff"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 64 sample have &gt;= 4 values for FPCA; removing 0 samples</span></span></code></pre></div>
<p>The modeling function will return the model as well as a message
letting the user know how many samples have been removed due to not
having enough radii with the spatial summary function calculated. In
order to get the model out of the <code>mxFDA</code> object to easily
plot, the <code><a href="../../reference/extract_model.html">extract_model()</a></code> function can be used with the
name provided in the above code block to store the model under and the
type of model (‘sofr’) that we ran. Looking at the class of the output
model, we can see that it has a class of “sofr” letting us know that it
is in-fact a scalar-on-function regression model.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">model</span> <span class="op">=</span> <span class="fu"><a href="../../reference/extract_model.html">extract_model</a></span><span class="op">(</span><span class="va">mxFDAobject</span>, <span class="st">'uni g'</span>, type <span class="op">=</span> <span class="st">'sofr'</span>, model_name <span class="op">=</span> <span class="st">'fit_sofr_tissue'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">model</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "sofr" "pfr"  "gam"  "glm"  "lm"</span></span></code></pre></div>
<p>By plotting this model, we can see the <span class="math inline">\(\hat{\beta}(r)\)</span> values for all of the
radii in column <code>r</code> of our data scaled to between 0 and 1.
The dashed lines show the 95% confidence intervals for the fit and where
those intervals no longer include 0 indicate where there is
statistically significant differences in CD8 T cell clustering between
our tumor and stromal compartments. Since this is a binary outcome and
not continuous, the <span class="math inline">\(\hat{\beta}(r)\)</span>
values are interpreted as the log odds ratio at radius <em>r</em>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">model</span>, ylab<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Special.html" class="external-link">beta</a></span><span class="op">(</span><span class="va">t</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, xlab<span class="op">=</span><span class="st">"t"</span><span class="op">)</span></span></code></pre></div>
<p><img src="st_fpca_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-insitutype" class="csl-entry">
Danaher, Patrick, Edward Zhao, Zhi Yang, David Ross, Mark Gregory, Zach
Reitz, Tae K. Kim, et al. 2022. <span>“Insitutype: Likelihood-Based Cell
Typing for Single Cell Spatial Transcriptomics.”</span>
<em>bioRxiv</em>. <a href="https://doi.org/10.1101/2022.10.19.512902" class="external-link">https://doi.org/10.1101/2022.10.19.512902</a>.
</div>
<div id="ref-increasedcoupling" class="csl-entry">
Soupir, Alex C, Mitchell T Hayes, Taylor C Peak, Oscar Ospina, Nicholas
H Chakiryan, Anders E Berglund, Paul A Stewart, et al. 2023.
<span>“Increased Spatial Coupling of Integrin and Collagen IV in the
Immunoresistant Clear Cell Renal Cell Carcinoma Tumor
Microenvironment.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2023.11.16.567457" class="external-link">https://doi.org/10.1101/2023.11.16.567457</a>.
</div>
<div id="ref-permutations" class="csl-entry">
Wilson, Christopher, Alex C. Soupir, Ram Thapa, Jordan Creed, Jonathan
Nguyen, Carlos Moran Segura, Travis Gerke, Joellen M. Schildkraut,
Lauren C. Peres, and Brooke L. Fridley. 2022. <span>“Tumor Immune Cell
Clustering and Its Association with Survival in African American Women
with Ovarian Cancer.”</span> <em>PLOS Computational Biology</em> 18 (3):
1–17. <a href="https://doi.org/10.1371/journal.pcbi.1009900" class="external-link">https://doi.org/10.1371/journal.pcbi.1009900</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Julia Wrobel, Alex Soupir.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
