<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="mxfda">
<title>mxFDA Object • mxfda</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="mxFDA Object">
<meta property="og:description" content="mxfda">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">mxfda</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/mx_fda.html">mxFDA Object</a>
    <a class="dropdown-item" href="../articles/mx_fpca.html">Multiplex FPCA</a>
    <a class="dropdown-item" href="../articles/mx_funreg.html">mx Functional Regression</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/julia-wrobel/mxfda/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>mxFDA Object</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/julia-wrobel/mxfda/blob/HEAD/vignettes/mx_fda.Rmd" class="external-link"><code>vignettes/mx_fda.Rmd</code></a></small>
      <div class="d-none name"><code>mx_fda.Rmd</code></div>
    </div>

    
    
<p>The <code>mxfda</code> package contains tools for analyzing
single-cell multiplex imaging data using methods from functional data
analysis. Analyses for this package are executed and stored using an S4
object of class <code>mxFDA</code>. This vignette outlines how to set up
an <code>mxFDA</code> object from spatial single cell imaging data, how
to calculate spatial summary functions, and exploratory data analysis
and visualization of these spatial summary functions. Details on how to
perform downstream analysis and feature extraction using functional
principal components analysis can be found in the separate vignette
<code><a href="../articles/mx_fpca.html">vignette("mx_fpca")</a></code>. To perform functional regression on
spatial summary functions from multiplex imaging data, see the
<code><a href="../articles/mx_funreg.html">vignette("mx_funreg")</a></code> vignette.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/julia-wrobel/mxfda" class="external-link">mxfda</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="vectrapolarisdata">VectraPolarisData<a class="anchor" aria-label="anchor" href="#vectrapolarisdata"></a>
</h2>
<p>Examples in this package use data adapted from the <a href="https://bioconductor.org/packages/release/data/experiment/html/VectraPolarisData.html" class="external-link">VectraPolarisData</a>
package on Bioconductor’s ExperimentHub. This package contains data from
two multiplex imaging experiments conducted at the University of
Colorado Anschutz Medical Campus. A shortcourse on single-cell multiplex
imaging using these data is available <a href="http://juliawrobel.com/MI_tutorial/MI_Data.html" class="external-link">here</a>.</p>
<p>Data has been preprocessed and stored directly in the
<code>mxfda</code> package. Available datasets are
<code>ovarian_FDA</code> and <code>lung_df</code>. This vignette will
focus on the <code>lung_df</code> dataset, which contains a subset of 50
subjects from a multiplex imaging study of non-small cell lung carcinoma
described in <span class="citation">Johnson et al. (2021)</span>. Each
subject has 4-5 multiplex images, which come from different regions of
interest (ROIs) in the tumor. Each ROI will be considered a “sample”
when constructing the <code>mxfda</code> object. We load the lung cancer
data below.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">lung_df</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="setting-up-the-mxfda-object">Setting up the mxFDA object<a class="anchor" aria-label="anchor" href="#setting-up-the-mxfda-object"></a>
</h2>
<p>The central object used with the <code>mxfda</code> package is the
<code>mxFDA</code> object. These objects are created with
<code><a href="../reference/make_mxfda.html">make_mxfda()</a></code> and hold everything from the raw spatial data
to fit functional data models using derived spatial summary functions.
To save space on large samples, the metadata is kept separately from the
spatial data and when needed, is exported and merged together. Slots in
the <code>mxFDA</code> object are designated as follows:</p>
<ul>
<li>
<code>Metadata</code> - stores sample specific traits that may be
used as covariates when fitting models</li>
<li>
<code>Spatial</code> - a data frame of cell level information (x and
y spatial coordinates, phenotype, etc.) that can be used to calculate
spatial summary functions</li>
<li>
<code>subject_key</code> - a character string for the column in the
metadata that denotes the unique subject id</li>
<li>
<code>sample_key</code> - a character string for the column in the
metadata that denotes the unique sample id. Note that there may be
multiple samples per subject, and this id links the metadata and spatial
data for each multiplex image sample</li>
<li>
<code>univariate_summaries</code> and
<code>bivariate_summaries</code> - lists of spatial summary functions
either imported with <code><a href="../reference/add_summary_function.html">add_summary_function()</a></code> or calculated
with <code><a href="../reference/extract_summary_functions.html">extract_summary_functions()</a></code>
</li>
<li>
<code>functional_pca</code> - list of results from functional
principle component analysis</li>
<li>
<code>functional_mpca</code> - list of results from multilevel
functional principle component analysis</li>
<li>
<code>functional_cox</code> - list of functional cox models that
have been fit</li>
<li>
<code>functional_mcox</code> - list of functional mixed cox models
that have been fit</li>
</ul>
<div class="section level3">
<h3 id="make-mxfdaobject">Make mxFDAobject<a class="anchor" aria-label="anchor" href="#make-mxfdaobject"></a>
</h3>
<p>Using the <code><a href="../reference/lung_df.html">?lung_df</a></code> in the <code>mxfda</code> package,
columns with repeated data pertaining to the sample-level information
are extracted and stored in a data frame called <code>clinical</code>
while the cell-level information is kept in a long data frame
(<code><a href="../reference/make_mxfda.html">make_mxfda()</a></code> also accepts cell-level information as a list
of data). The spatial parameter in the <code><a href="../reference/make_mxfda.html">make_mxfda()</a></code>
function can be left blank if using a spatial metric derived from
external functions. The final 2 parameters for the <code><a href="../reference/lung_df.html">?lung_df</a></code>
<code>mxFDA</code> object are the <code>subject_key</code> and the
<code>sample_key</code>. The <code>sample_key</code> is a column name
that appears both in the <code>metadata</code> and <code>spatial</code>
and denotes unique samples while the <code>subject_key</code> is a
column name in <code>metadata</code> that ties the samples to metadata;
if the data contains one sample per subject then <code>sample_key</code>
and <code>subject_key</code> is a 1:1, but if multiple samples per
subject, <code>subject_id</code> will be repeated.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clinical</span> <span class="op">=</span> <span class="va">lung_df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">image_id</span>, <span class="va">patient_id</span>, <span class="va">patientImage_id</span>, <span class="va">gender</span>, <span class="va">age</span>, <span class="va">survival_days</span>, <span class="va">survival_status</span>, <span class="va">stage</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spatial</span> <span class="op">=</span> <span class="va">lung_df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">image_id</span>, <span class="op">-</span><span class="va">gender</span>, <span class="op">-</span><span class="va">age</span>, <span class="op">-</span><span class="va">survival_days</span>, <span class="op">-</span><span class="va">survival_status</span>, <span class="op">-</span><span class="va">stage</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mxFDAobject</span> <span class="op">=</span> <span class="fu"><a href="../reference/make_mxfda.html">make_mxfda</a></span><span class="op">(</span>metadata <span class="op">=</span> <span class="va">clinical</span>,</span>
<span>                         spatial <span class="op">=</span> <span class="va">spatial</span>,</span>
<span>                         subject_key <span class="op">=</span> <span class="st">"patient_id"</span>,</span>
<span>                         sample_key <span class="op">=</span> <span class="st">"patientImage_id"</span><span class="op">)</span></span></code></pre></div>
<p>Note that the object created has class <code>mxFDA</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">mxFDAobject</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "mxFDA"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "mxfda"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="spatial-summary-functions-based-on-point-processes">Spatial summary functions based on point processes<a class="anchor" aria-label="anchor" href="#spatial-summary-functions-based-on-point-processes"></a>
</h2>
<p>The <code>mxfda</code> package provides methods for analyzing spatial
relationships between cell types in multiplex imaging data based on
point process theory. The location of cells in multiplex image samples
are treated as following a point process, realizations of a point
process are called “point patterns”, and point process models seek to
understand correlations in the spatial distributions of cells. Under the
assumption that the rate of a cell is constant over an entire region of
interest a point pattern will exhibit complete spatial randomness (CSR),
and it is often of interest to model whether cells deviate from CSR
either through clustering or repulsion. Spatial summary statistics can
be calculated to quantify the clustering and co-occurrence of cells in a
circular region with a particular radius <em>r</em>. Typically
univariate (one cell type) or bivariate (two cell types) summary
statistics are reported, and inference is obtained by comparing the
observed spatial summary statistic to that obtained under CSR. A popular
quantity is Ripley’s K(r), which studies the number of neighbors to a
particular point within radius <em>r</em>, and has univariate and
bivariate implementations in the <a href="http://spatstat.org/" class="external-link">spatstat</a> package (<span class="citation">Baddeley, Rubak, and Turner (2015)</span>). Ripley’s K
is characterized by clustering or repulsion depending on whether it is
above or below the theoretical value of <span class="math inline">\(\pi
r^2\)</span>. Other spatial summary statistics analyze the distance to a
neighbor, and can be interpreted as probabilities of observing a
particular cell type within a radius <em>r</em>. One of these metrics,
G(<em>r</em>), or the nearest neighbor distance distribution, is the
cumulative distribution function of an exponential random variable. More
detailed overviews of spatial summary functions for multiplex imaging
data are provided in <span class="citation">C. M. Wilson et al.
(2021)</span> and <span class="citation">Wrobel, Harris, and Vandekar
(2023)</span>.</p>
<div class="section level3">
<h3 id="univariate-summary-functions">Univariate summary functions<a class="anchor" aria-label="anchor" href="#univariate-summary-functions"></a>
</h3>
<p>Below we calculate univariate Ripley’s K to summarize the spatial
relationship among immune cells in each image. The <code>mxfda</code>
package accomplishes this with the function
<code><a href="../reference/extract_summary_functions.html">extract_summary_functions()</a></code>. Either univariate or bivariate
can be calculated with this function depending on the choice supplied to
the <code>extract_func</code> argument. To calculate a univariate
spatial summary we supply <code>extract_univariate</code> to the
<code>extract_func</code> argument. The summary function that is
calculated depends on the function supplied to <code>summary_fun</code>
which is one of <code><a href="../reference/Kest.html">Kest()</a></code>, <code><a href="../reference/Gest.html">Gest()</a></code>, or
<code><a href="../reference/Lest.html">Lest()</a></code> from the <a href="http://spatstat.org/" class="external-link">spatstat.explore</a> package
(<code><a href="../reference/Kcross.html">Kcross()</a></code>, <code><a href="../reference/Gcross.html">Gcross()</a></code>, or <code><a href="../reference/Lcross.html">Lcross()</a></code>
for bivariate methods). Other options include supplying a vector of
radius values through <code>r_vec</code>, and the a specific edge
correction (see <span class="citation">Baddeley, Rubak, and Turner
(2015)</span>). We calculate the K function across a range of radii from
0 to 100 and use the isotropic (“iso”) edge correction.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mxFDAobject</span> <span class="op">=</span> <span class="fu"><a href="../reference/extract_summary_functions.html">extract_summary_functions</a></span><span class="op">(</span><span class="va">mxFDAobject</span>,</span>
<span>                                        extract_func <span class="op">=</span> <span class="va">extract_univariate</span>,</span>
<span>                                        summary_func <span class="op">=</span> <span class="va">Kest</span>,</span>
<span>                                        r_vec <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">100</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>                                        edge_correction <span class="op">=</span> <span class="st">"iso"</span>,</span>
<span>                                        markvar <span class="op">=</span> <span class="st">"immune"</span>,</span>
<span>                                        mark1 <span class="op">=</span> <span class="st">"immune"</span><span class="op">)</span></span></code></pre></div>
<p>Running this code will calculate univariate Ripley’s K function to
measure spatial clustering of immune cells for each sample, and will
store these spatial summary functions in the
<code>univariate_summaries</code> slot of the <code>mxFDAobject</code>.
To access this slot and view the extracted summary functions, type:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mxFDAobject</span><span class="op">@</span><span class="va">univariate_summaries</span><span class="op">$</span><span class="va">Kest</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 24,847 × 5</span></span></span>
<span><span class="co">#&gt;    patientImage_id     r sumfun    csr fundiff</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 2_1                 0      0   0       0   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 2_1                 1      0   3.14   -<span style="color: #BB0000;">3.14</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 2_1                 2      0  12.6   -<span style="color: #BB0000;">12.6</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 2_1                 3      0  28.3   -<span style="color: #BB0000;">28.3</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 2_1                 4      0  50.3   -<span style="color: #BB0000;">50.3</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 2_1                 5      0  78.5   -<span style="color: #BB0000;">78.5</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 2_1                 6      0 113.   -<span style="color: #BB0000;">113.</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 2_1                 7      0 154.   -<span style="color: #BB0000;">154.</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 2_1                 8      0 201.   -<span style="color: #BB0000;">201.</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 2_1                 9      0 254.   -<span style="color: #BB0000;">254.</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 24,837 more rows</span></span></span></code></pre></div>
<p>Note that the summaries are returned as a dataframe. The variable
<code>sumfun</code> is the estimated summary function value,
<code>csr</code> is the theoretical value under complete spatial
randomness, and <code>fundiff</code> =
<code>sumfun</code>-<code>csr</code>; in downstream analysis we will use
the <code>fundiff</code> covariate.</p>
<div class="section level4">
<h4 id="plotting-the-mxfda-object">Plotting the <code>mxFDA</code> object<a class="anchor" aria-label="anchor" href="#plotting-the-mxfda-object"></a>
</h4>
<p><a href="https://github.com/julia-wrobel/mxfda" class="external-link">mxfda</a> has S4 methods for visualization implemented via
the <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> function (see <code><a href="../reference/plot.mxFDA.html">?plot.mxFDA</a></code> for
details). The first argument is the <code>mxFDA</code> object followed
by a few options that depend on what plot output is desired. Here, the
we want to plot the univariate summary that we just calculated, which
was the K function. By passing in <code>what = 'uni k'</code>, the plot
function will extract the univariate K results. We also need to tell
<code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> what column is the y-axis which can be
<code>'sumfun'</code> for the observed value, <code>'csr'</code> for the
theoretical value of complete spatial randomness (CSR), or
<code>'fundiff'</code> which is the difference between the observed K
measure and the theoretical CSR. The output of <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> is a
<a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a> object which can then be easily added
to/manipulated as any ggplot plot would.</p>
<p><strong>NOTE: These are the columns when calculating using the
<code>extract_summary_function()</code> of <a href="https://github.com/julia-wrobel/mxfda" class="external-link">mxfda</a> but if
summary data is added from elsewhere with
<code><a href="../reference/add_summary_function.html">add_summary_function()</a></code> then those column names will have to
be used.</strong></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mxFDAobject</span>, y <span class="op">=</span> <span class="st">"fundiff"</span>, what <span class="op">=</span> <span class="st">"uni k"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 48 rows containing missing values (`geom_line()`).</span></span></code></pre></div>
<p><img src="mx_fda_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="bivariate-summary-functions">Bivariate summary functions<a class="anchor" aria-label="anchor" href="#bivariate-summary-functions"></a>
</h3>
<p>The <code>extract_summary_function()</code> function can also be used
to extract bivariate summaries comparing spatial clustering of 2 cell
types. We will look at relationship between T-cells and macrophages.
There are a few images that have fewer than 5 T-cells or macrophages,
which makes estimation of spatial summary functions less stable for
those images. To look at T-cells and macrophages, the data phenotypes
and cell locations have to be in long format so we first create a
variable with the cell types (<code>'phenotype'</code>) from the
<code>lung_df</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lung_df</span> <span class="op">=</span> <span class="va">lung_df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>phenotype <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html" class="external-link">case_when</a></span><span class="op">(</span><span class="va">phenotype_cd8</span> <span class="op">==</span> <span class="st">"CD8+"</span> <span class="op">~</span> <span class="st">"T-cell"</span>,</span>
<span>                               <span class="va">phenotype_cd14</span> <span class="op">==</span> <span class="st">"CD14+"</span> <span class="op">~</span> <span class="st">"macrophage"</span>,</span>
<span>                               <span class="cn">TRUE</span> <span class="op">~</span> <span class="st">"other"</span><span class="op">)</span>,</span>
<span>         phenotype <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">phenotype</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We then recreate the mxFDAobject</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">spatial</span> <span class="op">=</span> <span class="va">lung_df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">image_id</span>, <span class="op">-</span><span class="va">gender</span>, <span class="op">-</span><span class="va">age</span>, <span class="op">-</span><span class="va">survival_days</span>, <span class="op">-</span><span class="va">survival_status</span>, <span class="op">-</span><span class="va">stage</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mxFDAobject</span> <span class="op">=</span> <span class="fu"><a href="../reference/make_mxfda.html">make_mxfda</a></span><span class="op">(</span>metadata <span class="op">=</span> <span class="va">clinical</span>,</span>
<span>                         spatial <span class="op">=</span> <span class="va">spatial</span>,</span>
<span>                         subject_key <span class="op">=</span> <span class="st">"patient_id"</span>,</span>
<span>                         sample_key <span class="op">=</span> <span class="st">"patientImage_id"</span><span class="op">)</span></span></code></pre></div>
<p>Now we calculate the bivariate G function, but can replace
<code><a href="../reference/Gcross.html">Gcross()</a></code> with <code><a href="../reference/Lcross.html">Lcross()</a></code> or
<code><a href="../reference/Kcross.html">Kcross()</a></code> to estimate the L or K bivariate functions
instead. The argument <code>markvar</code> takes the variable that we
created above called <code>'phenotype'</code>, and the 2 cell types that
we are interested in calculating the bivariate G for are
<code>'T-cell'</code> and <code>'macrophage'</code> so we provide them
to <code>mark1</code> and <code>mark2</code>, respectively.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mxFDAobject</span> <span class="op">=</span> <span class="fu"><a href="../reference/extract_summary_functions.html">extract_summary_functions</a></span><span class="op">(</span><span class="va">mxFDAobject</span>,</span>
<span>                summary_func <span class="op">=</span> <span class="va">Gcross</span>,</span>
<span>                extract_func <span class="op">=</span> <span class="va">extract_bivariate</span>,</span>
<span>                r_vec <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">100</span>, by <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>                edge_correction <span class="op">=</span> <span class="st">"rs"</span>,</span>
<span>                markvar <span class="op">=</span> <span class="st">"phenotype"</span>,</span>
<span>                mark1 <span class="op">=</span> <span class="st">"T-cell"</span>,</span>
<span>                mark2 <span class="op">=</span> <span class="st">"macrophage"</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="plotting-the-bivariate-g">Plotting the bivariate G<a class="anchor" aria-label="anchor" href="#plotting-the-bivariate-g"></a>
</h4>
<p>Just like with the univariate plots, we can use the
<code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot()</a></code> function to plot our <code>mxFDA</code> object
results. The <code>what</code> now is <code>'bi g'</code>,
<code>'bivar g'</code>, or <code>'bivariate g'</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mxFDAobject</span>, y <span class="op">=</span> <span class="st">"fundiff"</span>, what <span class="op">=</span> <span class="st">"bi g"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_hline</a></span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1365 rows containing missing values (`geom_line()`).</span></span></code></pre></div>
<p><img src="mx_fda_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="exploring-the-s4-object">Exploring the S4 object<a class="anchor" aria-label="anchor" href="#exploring-the-s4-object"></a>
</h2>
<p>Another useful function is <code><a href="../reference/summary.mxFDA.html">?summary.mxFDA</a></code> which feeds
into the <code><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary()</a></code> method. Either typing the name of the
object or wrapping it in the summary function will provide information
like the number of subjects, samples, if spatial summary functions have
been calculated, and functional data analyses that have been run.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mxFDAobject</span></span>
<span><span class="co">#&gt; mxFDA Object:</span></span>
<span><span class="co">#&gt;  Subjects: 50</span></span>
<span><span class="co">#&gt;  Samples: 247</span></span>
<span><span class="co">#&gt;  Has spatial data</span></span>
<span><span class="co">#&gt;  Univariate Summaries: None</span></span>
<span><span class="co">#&gt;  Bivariate Summaries: Gcross</span></span>
<span><span class="co">#&gt; FPCs not yet calculated</span></span>
<span><span class="co">#&gt; MFPCs not yet calculated</span></span>
<span><span class="co">#&gt; FCMs not yet calculated</span></span>
<span><span class="co">#&gt; MFCMs not yet calculated</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatialtime">SpatialTIME<a class="anchor" aria-label="anchor" href="#spatialtime"></a>
</h2>
<p>Sometimes other summary functions or normalizations are run outside
of the <a href="https://github.com/julia-wrobel/mxfda" class="external-link">mxfda</a> package but the end goal is to still run
functional data analysis. Other packages, such as
<a href="https://github.com/FridleyLab/spatialTIME" class="external-link">spatialTIME</a> (<span class="citation">Creed et al.
(2021)</span>) provide methods for fast calculation of functions in
<a href="http://spatstat.org/" class="external-link">spatstat</a> with permutation estimates of complete spatial
randomness that is more robust than the theoretical, especially when
tissue samples have holes that violate the assumption of a homogeneous
point pattern (see <span class="citation">C. Wilson et al.
(2022)</span>). Lets look at how to perform the estimation of univariate
nearest neighbor G with <a href="https://github.com/FridleyLab/spatialTIME" class="external-link">spatialTIME</a>.</p>
<p>The central object of <code>spatialTIME</code> is the
<code>mIF</code> object, that contains a list of spatial data frames, a
data frame of sample-level summaries, and a data frame for the metadata
(‘clinical’). From creating the <code>mxFDA</code> object, we have a
spatial data frame and the clinical data, now we have to convert them
into something that works with <code>spatialTIME.</code> The steps below
will be:</p>
<ol style="list-style-type: decimal">
<li>convert positive/negative to 1/0 integers</li>
<li>identify which columns in the spatial data frame are cell types</li>
<li>convert spatial data frame to a list, where each element is a unique
sample</li>
<li>create a summary data frame from the spatial list for the number and
proportion of positive cells for each phenotype</li>
</ol>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Step 1</span></span>
<span><span class="va">spatialTIME_spatial_df</span> <span class="op">=</span> <span class="va">spatial</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">X.1</span>, <span class="op">-</span><span class="va">X</span>, <span class="op">-</span><span class="va">phenotype</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="va">phenotype_ck</span><span class="op">:</span><span class="va">phenotype_cd4</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"\\+"</span>, <span class="va">.x</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html" class="external-link">relocate</a></span><span class="op">(</span><span class="va">patientImage_id</span>, .before <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Step 2</span></span>
<span><span class="va">cell_types</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">spatialTIME_spatial_df</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"phenotype"</span>, <span class="va">.</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Step 3</span></span>
<span><span class="va">spatial_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">spatialTIME_spatial_df</span>, <span class="va">spatial</span><span class="op">$</span><span class="va">patientImage_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Step 4</span></span>
<span><span class="va">summary_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">spatial_list</span>, <span class="kw">function</span><span class="op">(</span><span class="va">df</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">df</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="co">#group by sample ID to maintain ID column</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">patient_id</span>, <span class="va">patientImage_id</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="co">#find number of positive</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/reframe.html" class="external-link">reframe</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="op">!</span><span class="op">!</span><span class="va">cell_types</span>, <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span>,</span>
<span>              `Total Cells` <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="co">#calculate proportion</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="op">!</span><span class="op">!</span><span class="va">cell_types</span>, <span class="op">~</span><span class="va">.x</span><span class="op">/</span><span class="va">`Total Cells`</span> <span class="op">*</span> <span class="fl">100</span>, .names <span class="op">=</span> <span class="st">"{.col} %"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="co">#bind the rows together</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">bind_rows</span>, <span class="va">.</span><span class="op">)</span></span></code></pre></div>
<p>With the spatial list, clinical, and summary data the
<code>mIF</code> object can be constructed. For best computation
efficiency, use &gt;v1.3.4.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/FridleyLab/spatialTIME" class="external-link">spatialTIME</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; spatialTIME version:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBBBB;">1.3.4.1</span></span></span>
<span><span class="co">#&gt; If using for publication, please cite our manuscript:</span></span>
<span><span class="co">#&gt; <span style="color: #BBBBBB;">https://doi.org/10.1093/bioinformatics/btab757</span></span></span>
<span></span>
<span><span class="co">#make mif</span></span>
<span><span class="va">mif</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/spatialTIME/man/create_mif.html" class="external-link">create_mif</a></span><span class="op">(</span>clinical_data <span class="op">=</span> <span class="va">clinical</span>,</span>
<span>                 sample_data <span class="op">=</span> <span class="va">summary_data</span>,</span>
<span>                 spatial_list <span class="op">=</span> <span class="va">spatial_list</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span>,</span>
<span>                 patient_id <span class="op">=</span> <span class="st">"patient_id"</span>,</span>
<span>                 sample_id <span class="op">=</span> <span class="st">"patientImage_id"</span><span class="op">)</span></span></code></pre></div>
<p>Deriving spatial metrics with the <code>mIF</code> object is really
easy but does take some time. Will only do 10 permutations here to
estimate the complete spatial randomness measure of nearest neighbor G
and a reduced sampling, or <code>'rs'</code>, edge correction. To make
the run faster, will look at only cytotoxic T cells (CD8+) and helper T
cells (CD4+).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mif</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/spatialTIME/man/NN_G.html" class="external-link">NN_G</a></span><span class="op">(</span><span class="va">mif</span>, mnames <span class="op">=</span> <span class="va">cell_types</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span>,</span>
<span>           r_range <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">100</span>, num_permutations <span class="op">=</span> <span class="fl">10</span>, </span>
<span>           edge_correction <span class="op">=</span> <span class="st">"rs"</span>, keep_perm_dis <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>           workers <span class="op">=</span> <span class="fl">1</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span>, xloc <span class="op">=</span> <span class="st">"x"</span>, yloc <span class="op">=</span> <span class="st">"y"</span><span class="op">)</span></span></code></pre></div>
<p>With <code>spatialTIME</code>, all cell types (markers) are added to
the data frame. We can visualize both CD8+ and CD4+ with ggplot.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mif</span><span class="op">$</span><span class="va">derived</span><span class="op">$</span><span class="va">univariate_NN</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">r</span>, y <span class="op">=</span> <span class="va">`Degree of Clustering Permutation`</span>, color <span class="op">=</span> <span class="va">patientImage_id</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="op">~</span><span class="va">Marker</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1086 rows containing missing values (`geom_line()`).</span></span></code></pre></div>
<p><img src="mx_fda_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>Exporting the spatial summary function data from the <code>mIF</code>
object is the same as accessing the list object. However, we need to
make sure that the data that we use with the <code>mxFDA</code> object
contains only a single cell types results. This is to make sure that
when modeling we aren’t mixing up different cells. Below is the
extraction and filtering of the new univariate G results and keeping
only the cytotoxic T cell results</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">uni_g</span> <span class="op">=</span> <span class="va">mif</span><span class="op">$</span><span class="va">derived</span><span class="op">$</span><span class="va">univariate_NN</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"cd8"</span>, <span class="va">Marker</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>With the derived univariate nearest neighbor G for CD8+, it can be
added to an <code>mxFDA</code> object with
<code><a href="../reference/add_summary_function.html">add_summary_function()</a></code>. To show this, first will create a
new <code>mxFDA</code> object with an empty spatial slot then add the
new summary function results.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#make mxFDA object </span></span>
<span><span class="va">mxFDA_spatialTIME</span> <span class="op">=</span> <span class="fu"><a href="../reference/make_mxfda.html">make_mxfda</a></span><span class="op">(</span>metadata <span class="op">=</span> <span class="va">clinical</span>,</span>
<span>                               spatial <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                               subject_key <span class="op">=</span> <span class="st">"patient_id"</span>,</span>
<span>                               sample_key <span class="op">=</span> <span class="st">"patientImage_id"</span><span class="op">)</span></span>
<span><span class="co">#add summary data</span></span>
<span><span class="va">mxFDA_spatialTIME</span> <span class="op">=</span> <span class="fu"><a href="../reference/add_summary_function.html">add_summary_function</a></span><span class="op">(</span><span class="va">mxFDAobject</span>,</span>
<span>                                         summary_function_data <span class="op">=</span> <span class="va">uni_g</span>,</span>
<span>                                         metric <span class="op">=</span> <span class="st">"uni g"</span><span class="op">)</span></span></code></pre></div>
<p>Can now use the <code>mxfda</code> plot method with the new data and
continue with analyses as would be done if using the internal
<code>extract_summary_function()</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">mxFDA_spatialTIME</span>, y <span class="op">=</span> <span class="st">"Degree of Clustering Permutation"</span>, what <span class="op">=</span> <span class="st">"uni g"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 480 rows containing missing values (`geom_line()`).</span></span></code></pre></div>
<p><img src="mx_fda_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-baddeley2015spatial" class="csl-entry">
Baddeley, Adrian, Ege Rubak, and Rolf Turner. 2015. <em>Spatial Point
Patterns: Methodology and Applications with r</em>. CRC press.
</div>
<div id="ref-creed2021" class="csl-entry">
Creed, Jordan H, Christopher M Wilson, Alex C Soupir, Christelle M
Colin-Leitzinger, Gregory J Kimmel, Oscar E Ospina, Nicholas H
Chakiryan, et al. 2021. <span>“<span class="nocase">spatialTIME and
iTIME: R package and Shiny application for visualization and analysis of
immunofluorescence data</span>.”</span> <em>Bioinformatics</em> 37 (23):
4584–86. <a href="https://doi.org/10.1093/bioinformatics/btab757" class="external-link">https://doi.org/10.1093/bioinformatics/btab757</a>.
</div>
<div id="ref-johnson2021cancer" class="csl-entry">
Johnson, Amber M, Jennifer M Boland, Julia Wrobel, Emily K Klezcko, Mary
Weiser-Evans, Katharina Hopp, Lynn Heasley, et al. 2021. <span>“Cancer
Cell-Specific Major Histocompatibility Complex II Expression as a
Determinant of the Immune Infiltrate Organization and Function in the
NSCLC Tumor Microenvironment.”</span> <em>Journal of Thoracic
Oncology</em> 16 (10): 1694–704.
</div>
<div id="ref-wilson2021challenges" class="csl-entry">
Wilson, Christopher M, Oscar E Ospina, Mary K Townsend, Jonathan Nguyen,
Carlos Moran Segura, Joellen M Schildkraut, Shelley S Tworoger, Lauren C
Peres, and Brooke L Fridley. 2021. <span>“Challenges and Opportunities
in the Statistical Analysis of Multiplex Immunofluorescence
Data.”</span> <em>Cancers</em> 13 (12): 3031.
</div>
<div id="ref-wilson2022tumor" class="csl-entry">
Wilson, Christopher, Alex C Soupir, Ram Thapa, Jordan Creed, Jonathan
Nguyen, Carlos Moran Segura, Travis Gerke, Joellen M Schildkraut, Lauren
C Peres, and Brooke L Fridley. 2022. <span>“Tumor Immune Cell Clustering
and Its Association with Survival in African American Women with Ovarian
Cancer.”</span> <em>PLoS Computational Biology</em> 18 (3): e1009900.
</div>
<div id="ref-wrobel2023statistical" class="csl-entry">
Wrobel, Julia, Coleman Harris, and Simon Vandekar. 2023.
<span>“Statistical Analysis of Multiplex Immunofluorescence and
Immunohistochemistry Imaging Data.”</span> In <em>Statistical
Genomics</em>, 141–68. Springer.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Julia Wrobel, Alex Soupir.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
